sem/database.py

```diff
@@ -206,6 +207,16 @@ def insert_results(self, results):
             'meta': {k: ['...'] for k in ['elapsed_time', 'id', 'exitcode']},
         }
 
+        # If logging is enabled, the results should have a log_component 
+        # dictionary inside the meta dictionary
+        if log_component is not None:
+            log_component_list = list(log_component.keys())
+            example_result['meta']['log_component'] = {
+                k: '...' for k in log_component_list
+            }
```

Comment:
> Here, I think it should be enough to add `log_component` as a key to the `meta` entry in the `example_result` above (as shown in the suggested change). If that works, this portion of the code can be removed.

The `DatabaseManager.have_same_structure()` recursively looks for keys if there are nested dictionaries present inside the parent dictionary. If we just want to add `log_component` as a key, then we might also need to modify this function to prevent it from recursively looking inside the `log_component` dictionary.

Reply:
Again, i'm not familiar with this part of the code but what davide suggest should work fine: we want log_component to be an entry, but we do not want all entries to have the same log components (i guess?). so just placing log_component in example_result should be fine, without specifying its internal structure.
@DvdMgr is this correct?

Reply:
> > Here, I think it should be enough to add `log_component` as a key to the `meta` entry in the `example_result` above (as shown in the suggested change). If that works, this portion of the code can be removed.
> 
> **The `DatabaseManager.have_same_structure()` recursively looks for keys** if there are nested dictionaries present inside the parent dictionary. If we just want to add `log_component` as a key, then we might also need to modify this function to prevent it from recursively looking inside the `log_component` dictionary.

Didn't realize that! Maybe, then, it makes sense to only use `have_same_structure` for the `params` entry of the result (and at this point, it might be best to only directly check the keys, not even using `have_same_structure`), and separately check the `meta` entry... This would make `have_same_structure` useless, so we could just remove it at that point.